docs/content/topics/developers.md

```diff
@@ -8,47 +8,49 @@ aliases:
   - /topics/developers.md
 ---
 
-TODO: update per v2
-
 # Developer Guide
 
 This document explains how to get started developing Brigade.
 
-Brigade is composed of numerous parts.  The following represent the core components:
+Brigade is composed of numerous parts.  The following represent the core
+components:
 
-- brigade-controller: The Kubernetes controller for delegating Brigade events
-- brigade-worker: The JavaScript runtime for executing `brigade.js` files. The
-  controller spawns these, though you can run one directly as well.
-- brigade-api: The REST API server for user interfaces
-- brigade-project: The Helm [chart][brigade-project-chart] for installing Brigade projects
-- brigade-vacuum: The stale build cleaner-upper (optional; enabled by default)
+- apiserver: Brigade's API server, where the majority of core Brigade
+  logic lives
+- scheduler: Schedules event workers and jobs on the underlying substrate
+- observer: Observes (and records) state changes in workers and jobs and
+  ultimately cleans them up from the underlying substrate
+- worker: The default runtime for executing `brigade.js/.ts` files.
+- logger: Event log aggregator, with Linux and Windows variants
 - brig: The Brigade CLI
-- git-sidecar: The code that runs as a sidecar in cluster to fetch Git repositories (optional; enabled by default)
+- git-initializer: The code that runs as a sidecar to fetch Git repositories
+  for vcs-enabled projects
 
-Additionally, there are several opt-in gateways that can be enabled via Helm chart values.  These are:
+Additionally, there are several opt-in gateways that can be deployed alongside
+Brigade.  These are:
 
- - Brigade GitHub App Gateway: The implementation of the GitHub App web hooks. It requires
-  the controller.
- - Generic Gateway: A generic gateway offering flexibility to create Brigade events from webhooks
-  originating from an arbitrary service/platform.
-- Container Registry Gateway: A gateway supporting container registry webhooks such as the ones emitted by
-DockerHub and ACR.
+ - Brigade GitHub Gateway: The implementation of GitHub App-based web hooks.
 
-Read up on all the gateways above, as well as others, in the [Gateways doc](./gateways.md).
+For more information around available gateways and developing your own,
+see the [Gateways doc](./gateways.md).
 
-This document covers environment setup, how to run functional tests and development of
-`brigade-controller` and `brigade-worker` components.
+This document covers environment setup, how to run tests and development of
+core components.
 
 ## Prerequisites
 
-- Minikube or [kind](https://github.com/kubernetes-sigs/kind) (running k8s versions 1.16.0+)
-- Docker
+- A local Kubernetes cluster, 1.16.0+.  We recommend [kind] or [minikube].
+- [Docker]
 - make
 
+[kind]: https://github.com/kubernetes-sigs/kind
+[minikube]: https://github.com/kubernetes/minikube
+[Docker]: https://www.docker.com/
+
 ## Clone the Repository In GOPATH
```

Comment:
I'm unsure of the extent to which this is still necessary if using a recent version of Go.

Reply:
Ok. Just checked some Go docs and tried this for myself.

As long as one is using Go 1.16+ and `GO111MODULE` is _not_ explicitly set to `off` then there's no need to clone this on the GOPATH.

imho, we should just say that we require Docker (recommended) _or_ Go 1.16+ and leave it at that.

Reply:
Updated in https://github.com/brigadecore/brigade/pull/1521/commits/668c6d963f1dd3c0e28f7520b39f7101de0319d6; PTAL